hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/SnapshotValidator.java

```diff
@@ -31,30 +31,33 @@ private SnapshotValidator() {
     /**
      * Validates a snapshot with the given id.
      *
-     * @param snapshotId snapshot ID or {@link JobExecutionRecord#NO_SNAPSHOT},
-     *      if snapshot id is not known
+     * @param snapshotId snapshot ID or {@link JobExecutionRecord#NO_SNAPSHOT}, if snapshot id is
+     *                  not known
      * @param jobIdString name and id of the job (for debug output)
      * @param map snapshot map to validate
      * @return the snapshot id of the snapshot being validated
      */
     static long validateSnapshot(long snapshotId, String jobIdString, IMap<Object, Object> map) {
         SnapshotValidationRecord validationRecord = (SnapshotValidationRecord) map.get(SnapshotValidationRecord.KEY);
         if (validationRecord == null) {
-            throw new JetException("State for " + jobIdString + " was supposed to be restored from '" + map.getName()
-                    + "', but that map doesn't contain the validation record: not an IMap with Jet snapshot or corrupted");
+            throw new JetException(String.format(
+                    "snapshot-%d doesn't exist or is damaged. Unable to restore the state for %s.",
```

Comment:
this doesn't actually include the name of the snapshot the user requested

Reply:
We don't have snapshot name here, only the map name. The map name can also be the name of one of the runtime maps. I prefer keeping the old message: it says what we know happend, not what might be the case.